internal/ignition/ignition_test.go

```diff
@@ -813,69 +816,154 @@ var _ = Describe("downloadManifest", func() {
+		It("parses a v31 config with v31 override as 3.1.0", func() {
+			merge, err := MergeIgnitionConfig([]byte(v31ignition), []byte(v31override))
+			Expect(err).ToNot(HaveOccurred())
 
-		config, err := ParseToLatest([]byte(merge))
-		Expect(err).ToNot(HaveOccurred())
-		Expect(config.Ignition.Version).To(Equal("3.1.0"))
+			config, err := ParseToLatest([]byte(merge))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(config.Ignition.Version).To(Equal("3.1.0"))
 
-		bytes, err := json.Marshal(config)
-		Expect(err).ToNot(HaveOccurred())
-		v31Config, _, err := config_31.Parse(bytes)
-		Expect(err).ToNot(HaveOccurred())
-		Expect(v31Config.Ignition.Version).To(Equal("3.1.0"))
-	})
+			bytes, err := json.Marshal(config)
+			Expect(err).ToNot(HaveOccurred())
+			v31Config, _, err := config_31.Parse(bytes)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(v31Config.Ignition.Version).To(Equal("3.1.0"))
+		})
 
-	It("parses a v31 config with v32 override as 3.2.0", func() {
-		merge, err := MergeIgnitionConfig([]byte(v31ignition), []byte(v32override))
-		Expect(err).ToNot(HaveOccurred())
+		It("parses a v31 config with v32 override as 3.2.0", func() {
+			merge, err := MergeIgnitionConfig([]byte(v31ignition), []byte(v32override))
+			Expect(err).ToNot(HaveOccurred())
 
-		config, err := ParseToLatest([]byte(merge))
-		Expect(err).ToNot(HaveOccurred())
-		Expect(config.Ignition.Version).To(Equal("3.2.0"))
+			config, err := ParseToLatest([]byte(merge))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(config.Ignition.Version).To(Equal("3.2.0"))
+
+			bytes, err := json.Marshal(config)
+			Expect(err).ToNot(HaveOccurred())
+			v32Config, _, err := config_32.Parse(bytes)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(v32Config.Ignition.Version).To(Equal("3.2.0"))
+		})
+
+		It("parses a v32 config with v31 override as 3.2.0", func() {
+			merge, err := MergeIgnitionConfig([]byte(v32ignition), []byte(v31override))
+			Expect(err).ToNot(HaveOccurred())
+
+			config, err := ParseToLatest([]byte(merge))
+			Expect(err).ToNot(HaveOccurred())
+			Expect(config.Ignition.Version).To(Equal("3.2.0"))
+
+			bytes, err := json.Marshal(config)
+			Expect(err).ToNot(HaveOccurred())
+			v32Config, _, err := config_32.Parse(bytes)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(v32Config.Ignition.Version).To(Equal("3.2.0"))
+		})
+
+		It("does not parse v99 config", func() {
+			_, err := ParseToLatest([]byte(v99ignition))
+			Expect(err.Error()).To(ContainSubstring("unsupported config version"))
+		})
 
-		bytes, err := json.Marshal(config)
-		Expect(err).ToNot(HaveOccurred())
-		v32Config, _, err := config_32.Parse(bytes)
-		Expect(err).ToNot(HaveOccurred())
-		Expect(v32Config.Ignition.Version).To(Equal("3.2.0"))
 	})
 
-	It("does not parse v99 config", func() {
-		_, err := ParseToLatest([]byte(v99ignition))
-		Expect(err.Error()).To(ContainSubstring("unsupported config version"))
+	Context("using MergeStructTranscribe", func() {
```

Comment:
I don't understand these tests. They look like they're just testing the ignition library, shouldn't these be using `MergeIgnitionConfig`?

Reply:
We have tests that use library (because my impression was that we seem to use library without fully understand it) and also tests that use `MergeIgnitionConfig` (just below)